src/core/tritonserver.cc

```diff
@@ -2400,17 +2421,6 @@ TRITONSERVER_ServerInferAsync(
   // Run inference...
   ni::Status status = lserver->InferAsync(ureq);
 
-  // If there is an error then must explicitly release any trace
-  // object associated with the inference request above.
-#ifdef TRITON_ENABLE_TRACING
-  if (!status.IsOk()) {
-    std::unique_ptr<ni::InferenceTrace>* trace = ureq->MutableTrace();
-    if (*trace != nullptr) {
-      ni::InferenceTrace::Release(std::move(*trace));
```

Comment:
You still need to "*trace = nullptr" here to release the reference to the trace object which will cause the trace release_fn_ to be called. Otherwise we are changing the existing behavior